scico/linop/_stack.py

```diff
@@ -11,19 +11,50 @@
 
 import operator
 from functools import partial
-from typing import List, Optional, Union
+from typing import List, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
+from typing_extensions import TypeGuard
+
 import scico.numpy as snp
 from scico.numpy import BlockArray
 from scico.numpy.util import is_nested
-from scico.typing import JaxArray
+from scico.typing import BlockShape, JaxArray, Shape
 
 from ._linop import LinearOperator, _wrap_add_sub, _wrap_mul_div_scalar
 
 
-class LinearOperatorStack(LinearOperator):
+def collapse_shapes(
+    shapes: Sequence[Union[Shape, BlockShape]], allow_collapse=True
+) -> Tuple[Union[Shape, BlockShape], bool]:
+    """Decides whether to collapse a sequence of shapes and returns the collpased
+    shape and a boolean indicating whether the shape was collapsed."""
+
+    if is_collapsable(shapes) and allow_collapse:
```

Comment:
A bit of a nitpick, but of the two variant spellings `collapsable` and `collapsible`, the latter seems to be much more common, having rank 2848 in [a database of English word frequencies](https://www.wordandphrase.info/frequencyList.asp), while the former doesn't appear at all.